src/JuliaVariables.jl

```diff
@@ -504,6 +504,23 @@ function solve(ast; toplevel=true)
     transform(ast)
 end # module struct
 
-solve_from_local(@nospecialize(ex)) = solve(ex; toplevel=false)
+solve_from_local!(@nospecialize(ex)) = solve!(ex; toplevel=false)
+
+_depwarn(sym) = Base.depwarn(
+    "`$sym(ex)` is deprecated in JuliaVariables 0.2.x. " *
+    "Please use `$(sym)!(ex)` instead.  `$sym(ex)`, which currently mutates `ex` " *
+    "in-place, will be re-introduced in 0.3 as a non-mutating variant.",
+    sym,
+)
```

Comment:
I wasn't sure if you want to go straight to 0.3 or stay at 0.2 a bit more.  But I guess deprecating `solve` for now while introducing `solve!` would be the most user-friendly thing to do at the moment?

Reply:
I won't go to 0.3 immediately, but it wouldn't take a long time.
> But I guess deprecating solve for now while introducing solve! would be the most user-friendly thing to do at the moment?

Right.